### DISCUSSION TITLE
Change the sign of flux value upon reaching the certain average value on the surface of the domain

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20178] [Q&A Modules: General] [answered]
Hello MOOSE Experts

I have been working on the thermo-mechanical model. I want to change the value of flux upon reaching a certain average value of temperature on the surface. Basically, I want to make it cyclic flux. 

I am using Neumann BC for applying flux and for calculating the average temperature on the surface using the "SideAverageValue" postprocessor module is considered.

Let's say, Initial value of Temp = 0, then apply flux and the average temperature reaches 100. At that point, I want to change the flux from +ve to -ve, keeping the magnitude the same. Again if it reaches 0, I want to change the flux value from -ve to +ve, while keeping the magnitude the same.

Can you please suggest to me the best way to do it?

I shall be very thankful.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

You should leverage the Control system to modify the flux value in the NeumannBC, when a given postprocessor reaches a certain value. 
https://mooseframework.inl.gov/syntax/Controls/index.html

You can create your own control off of the PID control to make a control that sets a value based on a postprocessor

Guillaume
